src/promptflow-core/promptflow/core/_prompty_utils.py

```diff
@@ -542,12 +642,27 @@ def parse_chat(chat_str, images: List = None, valid_roles: List[str] = None):
 
     for chunk in chunks:
         last_message = chat_list[-1] if len(chat_list) > 0 else None
-        if last_message and "role" in last_message and "content" not in last_message:
+        if is_tool_chunk(last_message):
```

Comment:
This is another new feature tools&tool_choice, might be inappropriate to name this PR title as Escape/unescape only. Please modify the PR title if you want to include the tools feature support in this PR.

Reply:
Besides, if you want to support tools, might need to process tool choices, refer to [code](https://github.com/microsoft/promptflow/blob/main/src/promptflow-tools/promptflow/tools/common.py#L826)